src/v/crypto/hmac.cc

```diff
@@ -18,22 +18,43 @@
 #include <openssl/params.h>
 
 namespace crypto {
+namespace {
+static const auto sha256_params = []() {
+    std::array<OSSL_PARAM, 2> params{
+      OSSL_PARAM_construct_utf8_string(
+        OSSL_MAC_PARAM_DIGEST, const_cast<char*>("SHA256"), 0),
+      OSSL_PARAM_construct_end()};
+    return params;
+}();
+
+static const auto sha512_params = []() {
+    std::array<OSSL_PARAM, 2> params{
+      OSSL_PARAM_construct_utf8_string(
+        OSSL_MAC_PARAM_DIGEST, const_cast<char*>("SHA512"), 0),
+      OSSL_PARAM_construct_end()};
+    return params;
+}();
+
+static const std::array<OSSL_PARAM, 2>& get_param(digest_type type) {
+    switch (type) {
+    case digest_type::SHA256:
+        return sha256_params;
+    case digest_type::SHA512:
+        return sha512_params;
+    default:
+        vassert(false, "NOPE");
```

Comment:
Since this isn't a covering set, the assertion is nice. But maybe print out what it is, like MD5 looks like it could be the problem.